cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -64,6 +64,22 @@ struct BinaryLength {
   }
 };
 
+struct Utf8Length {
+  template <typename OutValue, typename Arg0Value = util::string_view>
+  static OutValue Call(KernelContext*, Arg0Value val) {
+    auto str = reinterpret_cast<const uint8_t *>(val.data());
+    auto strlen = val.size();
+
+    OutValue length = 0;
+    while (strlen > 0) {
+      length += ((*str & 0xc0) != 0x80);
```

Comment:
It may be even simpler: `length += (*str < 0x80);`

Reply:
I did try this solution but it fails for multibyte characters, since they begin with the form 0xc0... and 0xc0 > 0x80. This approach only works for single byte UTF8. Please refer to UTF8 [encoding](https://tools.ietf.org/html/rfc3629#section-3).

Reply:
Oh, yes, you're right. Thanks for the correction.